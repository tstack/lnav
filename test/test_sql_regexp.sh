#! /bin/bash

export YES_COLOR=1

run_cap_test ${lnav_test} -n \
   -c ";SELECT * FROM syslog_log, regexp_capture_into_json(log_body, '"'"'"(?<value>[^"'"'"]+)')" \
   -c ":write-csv-to -" \
   ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -n \
   -c ";SELECT * from regexp_capture_into_json('foo=0x123e;', '(?<key>\w+)=(?<value>[^;]+)')" \
   ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -n \
   -c ";SELECT * from regexp_capture_into_json('foo=0x123e;', '(?<key>\w+)=(?<value>[^;]+)', json_object('convert-numbers', json('false')))" \
   ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -n \
   -c ";SELECT * from regexp_capture_into_json('foo=0x123e;', '(?<key>\w+)=(?<value>[^;]+)', '{abc')" \
   ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -n \
   -c ";SELECT * from regexp_capture_into_json('foo=123e;', '(?<key>\w+)=(?<value>[^;]+)')" \
   ${test_dir}/logfile_syslog.3

run_cap_test ${lnav_test} -nN \
   -c ";SELECT * from regexp_capture('abc=def;ghi=jkl;', '^(\w+)=([^;]+);')"

run_cap_test ${lnav_test} -nN \
   -c ";SELECT * from regexp_capture_into_json('abc=def;ghi=jkl;', '^(\w+)=([^;]+);')"
