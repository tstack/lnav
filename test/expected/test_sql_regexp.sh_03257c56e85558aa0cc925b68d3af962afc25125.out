[1m[4mmatch_index [0m[1m[4mcapture_index [0m[1m[4mcapture_name [0m[1m[4mcapture_count [0m[1m[4mrange_start [0m[1m[4mrange_stop [0m[1m[4mcontent  [0m
[1m          0[0m[1m [0m[1m            0[0m[1m [0m[1m      <[0m[1mNULL[0m[1m>[0m[1m [0m[1m            3[0m[1m [0m[1m          1[0m[1m [0m[1m         9[0m[1m [0m[1mabc=def; [0m
          0             1       <NULL>             3           1          4 abc      
[1m          0[0m[1m [0m[1m            2[0m[1m [0m[1m      <[0m[1mNULL[0m[1m>[0m[1m [0m[1m            3[0m[1m [0m[1m          5[0m[1m [0m[1m         8[0m[1m [0m[1mdef      [0m
